runtime/j9vm/java11vmi.c

```diff
@@ -417,7 +417,13 @@ addModuleDefinition(J9VMThread * currentThread, J9Module * fromModule, const cha
 	J9ClassLoader * const classLoader = fromModule->classLoader;
 
 	UDATA retval = ERRCODE_GENERAL_FAILURE;
-	if (!areNoPackagesDefined(currentThread, classLoader, packages, numPackages)) {
+
+	/* the bootstrap classloader is exempt from unnamed module checks
+	 * because it may place classes in the unnamed
+	 * module before java.base is created.
+	 */
+	if (!areNoPackagesDefined(currentThread, classLoader, packages, numPackages,
+			(classLoader != currentThread->javaVM->systemClassLoader) || !isModuleJavaBase(currentThread, fromModule->moduleName))) {
```

Comment:
Please don't inline this check into the function call.  It's much clearer to do the check ahead of time and store it in a local